backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -425,7 +425,11 @@ private void popStepInCurrentSessionFrame() {
     }
 
     public void setDatum(String keyId, String value) {
-        frame.pushStep(new StackFrameStep(SessionFrame.STATE_DATUM_VAL, keyId, value));
+        setDatum(SessionFrame.STATE_DATUM_VAL, keyId, value);
+    }
+
+    public void setDatum(String action, String keyId, String value) {
+        frame.pushStep(new StackFrameStep(action, keyId, value));
```

Comment:
@phillipm adding a test required changing the loadSessionFromDescriptor method - previously every datum was set as a CASE_ID so these ambiguities between unknown, computed, and case_id were lost